lib/msprime.c

```diff
@@ -2392,6 +2426,31 @@ msp_get_common_ancestor_waiting_time(msp_t *self, uint32_t population_id)
             (uint32_t) avl_count(&pop->ancestors));
 }
 
+static double
+msp_get_multiple_merger_waiting_time(msp_t *self, uint32_t population_id,
+        unsigned int k) // k used for k-merger
+{
+    double ret = DBL_MAX;
+    population_t *pop = &self->populations[population_id];
+    unsigned int n = avl_count(&pop->ancestors);
+    //double t = self->time;
+    double u;
+    double mm_rate = 0.0;
+
+    if (self->model.type == MSP_MODEL_DIRAC){
+        mm_rate = msp_compute_lambda_Xi_dirac(n,
+                      self->model.params.dirac_coalescent.psi,
+                      self->model.params.dirac_coalescent.const_c);
+    }
+
+    if (mm_rate > 0.0) {
+        u = gsl_ran_exponential(self->rng, 1.0/mm_rate)/2.0;
+        ret = pop->initial_size * u;
```

Comment:
Let's stay away from population structure for now. Assume 1 population with a fixed size of 1.